src/sentry/search/events/fields.py

```diff
@@ -325,6 +336,28 @@ def team_key_transaction_expression(organization_id, team_ids, project_ids):
 }
 
 
+class FieldAlias:
+    def __init__(self, expression_fn, return_type=None) -> None:
+        self.expression_fn = expression_fn
+        self.return_type = return_type
+
+    def get_column(self, params) -> SnqlFunction:
+        return self.expression_fn(params)
+
+
+SNQL_FIELD_ALIASES = {
+    # TODO: Remove the `toUInt64` once Column supports aliases
+    ISSUE_ALIAS: FieldAlias(
+        # NOTE: `ISSUE_ALIAS` simply maps to the id, meaning that post processing is required to
+        # insert the true issue short id into the response.
+        lambda _: SnqlFunction("toUInt64", [SnqlColumn("group_id")], "issue.id")
```

Comment:
Not really happy about this because we likely want to run `self.resolve_column_name` here to resolve `issue.id` into `group_id`.

Reply:
Yeah, I was leaning towards that. Though it may result in `QueryFields` being a rather large class.

Reply:
Ended up introducing `QueryFieldAliases` as a mixin that makes mapping field aliases to the SnQL columns simpler. It's also usable on it's own and can be introduced to `QueryFilter` for things like `user.display`.